tests/v2/validation/provisioning/k3s/secret_encryption_test.go

```diff
@@ -0,0 +1,320 @@
+			provider := CreateProvider(provider)
+			clusterResp, metadata := TestProvisioningK3SCluster(s.T(), client, provider, s.nodesAndRoles, k8sVersion, s.psact, s.advancedOptions, false)
+
+			clusterID, err := clusters.GetClusterIDByName(client, clusterResp.Name)
+			require.NoError(s.T(), err)
+
+			podResults, podErrors := pods.StatusPods(client, clusterID)
+			assert.NotEmpty(s.T(), podResults)
+			assert.Empty(s.T(), podErrors)
+
+			steveClient, err := client.Steve.ProxyDownstream(clusterID)
+			require.NoError(s.T(), err)
+			assert.NotEmpty(s.T(), steveClient)
+
+			secretTemplate := secrets.NewSecretTemplate(namegen.AppendRandomString("test-secret-"), s.ns, map[string][]byte{secretKey: []byte(secretValue)})
+			secretObj, err := steveClient.SteveType(secrets.SecretSteveType).Create(secretTemplate)
+			require.NoError(s.T(), err)
+			logrus.Infof("secret created successfully!")
+
+			envFrom := []v1.EnvFromSource{
+				{
+					SecretRef: &v1.SecretEnvSource{
+						LocalObjectReference: v1.LocalObjectReference{Name: secretObj.Name},
+					},
+				},
+			}
+			wload := namegen.AppendRandomString("test-workload-")
+			containerTemplate := workloads.NewContainer("ranchertest", "ranchertest/mytestcontainer", v1.PullAlways, []v1.VolumeMount{}, envFrom)
+			podTemplate := workloads.NewPodTemplate([]v1.Container{containerTemplate}, []v1.Volume{}, []v1.LocalObjectReference{}, nil)
+			deploymentTemplate := workloads.NewDeploymentTemplate(wload, s.ns, podTemplate, true, nil)
+
+			_, err = steveClient.SteveType(workloads.DeploymentSteveType).Create(deploymentTemplate)
+			require.NoError(s.T(), err)
+
+			podResults, podErrors = pods.StatusPods(client, clusterID)
+			assert.NotEmpty(s.T(), podResults)
+			assert.Empty(s.T(), podErrors)
+
+			pods, err := steveClient.SteveType(pods.PodResourceSteveType).List(nil)
+			require.NoError(s.T(), err)
+
+			kubeConfig, err := kubeconfig.GetKubeconfig(client, clusterID)
+			require.NoError(s.T(), err)
+
+			restConfig, err := (*kubeConfig).ClientConfig()
+			require.NoError(s.T(), err)
+
+			cmd := []string{
+				"/bin/sh",
+				"-c",
+				"echo $testKey",
+			}
+
+			for _, pod := range pods.Data {
+				if strings.HasPrefix(pod.ObjectMeta.Name, wload) {
+					err := kwait.Poll(5*time.Second, 2*time.Minute, func() (done bool, err error) {
+						output, err := kubeconfig.KubectlExec(restConfig, pod.ObjectMeta.Name, s.ns, cmd)
+						if err != nil {
+							return false, nil
+						}
+						require.Equal(s.T(), output.String(), secretValue)
+						logrus.Info("environment variable "+secretKey+" value : ", output.String())
+						return true, nil
+					})
+					require.NoError(s.T(), err)
+					break
+				}
+			}
+
+			nodesList, err := steveClient.SteveType("node").List(nil)
+			require.NoError(s.T(), err)
+			assert.NotEmpty(s.T(), nodesList)
+
+			client, err = client.ReLogin()
+			require.NoError(s.T(), err)
+
+			for _, node := range nodesList.Data {
+				if node.Labels[etcdLabel] == "true" {
```

Comment:
I think instead of string check; we can make `strconv.ParseBool()` here. Applicable for other string boolean checks.

Reply:
Okay, I've replaced the string check with strconv.ParseBool() now.